### ISSUE TITLE
Remote.fetch() fails.

### ISSUE THREAD

**Issue by StyXman** [gitpython-developers/GitPython#325]
Installed as `python-git` from Debian Testing (package version is 0.3.2~RC1-3):

```python
In [1]: import git

In [2]: repo= git.Repo.init ('merde')   

In [3]: pricing= repo.create_remote ('pricing', 'http://user@machine/git/scm/sspc/pricing.git') 

In [4]: pricing.fetch ()
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-4-1588473e7cee> in <module>()
----> 1 pricing.fetch ()

/usr/lib/python2.7/dist-packages/git/remote.pyc in fetch(self, refspec, progress, **kwargs)
    591                 kwargs = add_progress(kwargs, self.repo.git, progress)
    592                 proc = self.repo.git.fetch(self, refspec, with_extended_output=True, as_process=True, v=True, **kwargs)
--> 593                 return self._get_fetch_info_from_stderr(proc, progress or RemoteProgress())
    594 
    595         def pull(self, refspec=None, progress=None, **kwargs):

/usr/lib/python2.7/dist-packages/git/remote.pyc in _get_fetch_info_from_stderr(self, proc, progress)
    537                 fp.close()
    538 
--> 539                 assert len(fetch_info_lines) == len(fetch_head_info), "len(%s) != len(%s)" % (fetch_head_info, fetch_info_lines)
    540 
    541                 output.extend(FetchInfo._from_line(self.repo, err_line, fetch_line) 

AssertionError: len(["5487c312cef344b6709a293ba55450861fa2105e\tnot-for-merge\tbranch 'PTR6472300' of http://rndwww.nce.amadeus.net/git/scm/sspc/pricing\n", ...
```

I cut off the rest because it's lines and lines of output. I'm not sure if you need access to the problematic repo, but given that I can run `git fetch` from it and that I can't actually give you such access, we'll have to work from here.
Let me know what data can I give you from the result of this (there *is* data in the resulting repo, just not sure how consistent).

**Comment by Byron** [comment]
Thanks for the report.

This is a very old version and chances are that this is already fixed in version v1.0.1 from Pypi or (if not) the master branch from this repository.

Please try a newer version to see if it fixes the problem.

If you are unable to get the latest version, and if you are not interested in the return value of `pricing.fetch()`, you may as well call git directly using `repo.git.fetch(pricing)`. That will definitely work and should be a suitable workaround.

Please post your findings here as you see fit.

**Comment by StyXman** [comment]
Both the workaround you mention and the latest version work. Closing.
